File: backend/controller/answerController.js
const dbConnection = require("../db/dbConfig")
const { StatusCodes } = require("http-status-codes")

// Get answers for a specific question
const getAnswer = async (req, res) => {
    try {
        const { questionid } = req.params
        const [answers] = await dbConnection.query(
            `SELECT a.answerid, a.userid, a.questionid, a.answer_text as answer, u.username 
             FROM answers a 
             JOIN users u ON a.userid = u.userid 
             WHERE a.questionid = ?`,
            [questionid]
        )
        
        console.log("Found answers:", answers)
        
        if (!answers || answers.length === 0) {
            console.log("No answers found for question:", questionid)
            return res.status(StatusCodes.OK).json({ 
                message: "No answers found", 
                answers: [] 
            })
        }

        return res.status(StatusCodes.OK).json({ 
            message: "Answers retrieved successfully", 
            answers 
        })
    } catch (error) {
        console.error("Error getting answers:", error)
        return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ 
            message: "An unexpected error occurred",
            error: error.message 
        })
    }
}

// Post a new answer
const postAnswer = async (req, res) => {
    const { questionid, answer } = req.body
    const userid = req.user.userid // Get userid from auth middleware
    console.log("Received post answer request:", { questionid, answer, userid })
    
    if (!answer) {
        return res
            .status(StatusCodes.BAD_REQUEST)
            .json({ message: "Please provide answer" })
    }

    try {
        const [result] = await dbConnection.query(
            "INSERT INTO answers (userid, questionid, answer_text) VALUES (?, ?, ?)",
            [userid, questionid, answer]
        )

        console.log("Answer posted successfully:", result)

        return res
            .status(StatusCodes.CREATED)
            .json({ message: "Answer posted successfully" })
    } catch (error) {
        console.error("Error posting answer:", error)
        return res
            .status(StatusCodes.INTERNAL_SERVER_ERROR)
            .json({ 
                message: "An unexpected error occurred",
                error: error.message 
            })
    }
}

// Get answer statistics
const getAnswerStats = async (req, res) => {
    try {
        const [[{ totalAnswers }]] = await dbConnection.query(
            "SELECT COUNT(*) AS totalAnswers FROM answers"
        )

        res.status(StatusCodes.OK).json({
            totalAnswers,
        })
    } catch (error) {
        console.error("Get answer count error:", error)
        res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
            msg: "Error fetching answer count",
            error: error.message,
        })
    }
}

module.exports = {
    getAnswer,
    postAnswer,
    getAnswerStats
}

File: Backend/controller/AnswerController.kt
package controllers

import io.ktor.application.*
import io.ktor.http.HttpStatusCode
import io.ktor.response.respond
import io.ktor.request.receive
import io.ktor.routing.*
import io.ktor.server.engine.embeddedServer
import io.ktor.server.netty.Netty
import db.DatabaseFactory
import io.ktor.features.ContentNegotiation
import io.ktor.serialization.kotlinx.json
import kotlinx.serialization.Serializable

// Answer model for the response
@Serializable
data class Answer(val answerId: String, val userId: String, val questionId: String, val answer: String, val username: String)

suspend fun postAnswer(call: ApplicationCall) {
    val params = call.receive<Map<String, String>>()
    val answer = params["answer"]
    val questionId = params["questionid"]
    val userId = params["userid"]

    if (answer == null) {
        call.respond(HttpStatusCode.BadRequest, mapOf("message" to "Please provide answer"))
        return
    }

    try {
        DatabaseFactory.dbQuery {
            // Insert answer into the database
            val query = "INSERT INTO answers (userid, questionid, answer) VALUES (?, ?, ?)"
            // db.execute(query, userId, questionId, answer)
        }
        call.respond(HttpStatusCode.Created, mapOf("message" to "Answer posted successfully"))
    } catch (e: Exception) {
        call.respond(HttpStatusCode.InternalServerError, mapOf("message" to "An unexpected error occurred."))
    }
}

suspend fun getAnswer(call: ApplicationCall) {
    val questionId = call.parameters["questionid"]

    try {
        val answers = DatabaseFactory.dbQuery {
            // Execute query and retrieve answers
            val query = """
                SELECT a.answerid, a.userid, a.questionid, a.answer, u.username 
                FROM answers a 
                JOIN users u ON a.userid = u.userid 
                WHERE questionid = ?
            """
            // Execute query with questionId
            // val result = db.query(query, questionId)
            // Return result as a list of Answer
        }
        call.respond(HttpStatusCode.OK, mapOf("message" to "Answers retrieved successfully", "answers" to answers))
    } catch (e: Exception) {
        call.respond(HttpStatusCode.InternalServerError, mapOf("message" to "An unexpected error occurred."))
    }
}

suspend fun getAnswerStats(call: ApplicationCall) {
    try {
        val totalAnswers = DatabaseFactory.dbQuery {
            // Execute query to count total answers
            val query = "SELECT COUNT(*) AS totalAnswers FROM answers"
            // db.query(query)
            // Return total answers count
        }

        call.respond(HttpStatusCode.OK, mapOf("totalAnswers" to totalAnswers))
    } catch (e: Exception) {
        call.respond(HttpStatusCode.InternalServerError, mapOf("message" to "Error fetching answer count"))
    }
}
